allennlp/tests/nn/beam_search_test.py

```diff
@@ -167,3 +167,24 @@ def test_catch_bad_config(self):
         beam_search = BeamSearch(self.end_index, beam_size=20)
         with pytest.raises(ConfigurationError):
             self._check_results(beam_search=beam_search)
+
+    def test_warn_for_bad_log_probs(self):
+        # The only valid next step from the initial predictions is the end index.
+        # But with a beam size of 3, the call to `topk` to find the 3 most likely
+        # next beams will result in 2 new beams that are invalid, in that have probability of 0.
+        # The beam search should warn us of this.
+        initial_predictions = torch.LongTensor([self.end_index-1, self.end_index-1])
+        with pytest.warns(RuntimeWarning):
```

Comment:
Please use something like `pytest.warns(RuntimeWarning, match="foo")` to distinguish between this and the case below.